source/common/config/BUILD

```diff
@@ -22,6 +22,31 @@ envoy_cc_library(
     ],
 )
 
+# Pack API type database text file into a char* string that can be referenced
+# at the C++ level.
+genrule(
+    name = "api_type_db_genrule",
+    srcs = ["api_type_db.generated.pb_text"],
+    outs = ["api_type_db_def.generated.cc"],
+    cmd = """
```

Comment:
I'm a bit worried about the code size by embedding in proto text format, shall we use binary format? (AFAIK protobuf embed descriptors in binary format.)

Reply:
We could as a small optimization. I had a look at the difference between `.pb` and `.pb_text` previously, when I was making the change in the previous PR. The (long) type names dominate, so I think you don't get a ton of savings, it's less than 2x.

Reply:
I'm late to this, but code size does matter a lot to us in Envoy Mobile. Optimally, we would even be able to compile out the translation support entirely if we were to explicitly say that we don't want to support the previous version for example. If this is easy I would throw that out there is a feature request. cc @junr03 @goaway

Reply:
I have some ideas how to specialize this, I'm going to close this one out for now, as I want to put together a design doc on API boosting first so we all have directional alignment.